tests/adapters/test_adapter.py

```diff
@@ -62,15 +110,33 @@ def test_not_implemented_methods(adapter, method, args):
                           ))
 def test_adapter_values(value, kwargs, result):
     a = FakeAdapter()
-    assert a.values(value, **kwargs) == result
+    with pytest.warns(FutureWarning):
+        assert a.values(value, **kwargs) == result
+
+
+def test_read_binary_values():
+    a = ProtocolAdapter([(None, "abcdefgh")])
+    assert list(a.read_binary_values()) == pytest.approx([1.6777999e+22, 4.371022e+24])
+
+
+def test_write_binary_values():
+    a = ProtocolAdapter([(b'CMD#212\x00\x00\x80?\x00\x00\x00@\x00\x00@@\n', None)])
+    a.write_binary_values("CMD", [1, 2, 3], termination="\n")
 
 
 def test_adapter_preprocess_reply():
-    a = FakeAdapter(preprocess_reply=lambda v: v[1:])
-    assert str(a) == "<FakeAdapter>"
-    assert a.values("R42.1") == [42.1]
-    assert a.values("A4,3,2") == [4, 3, 2]
-    assert a.values("TV 1", preprocess_reply=lambda v: v.split()[0]) == ['TV']
-    assert a.values("15", preprocess_reply=lambda v: v) == [15]
+    with pytest.warns(FutureWarning):
```

Comment:
should that test be restructured, to avoid the FutureWarning?

Reply:
we deprecated preprocess_reply in the Adapter.
The test will be removed, once we remove preprocess_reply, but right now, I like to test the method, as it is still in use.